### ISSUE TITLE
Run on Apple Mac Silicon chip

### ISSUE THREAD

**Issue by ncamcl** [SakanaAI/AI-Scientist#43]
I found that the current problem can only run on CPU  model on Apple silicon chip. I don't know if the GPU method which is 'mps' can be added.  I set the parameters as mps, but it didn't work.

**Comment by conglu1997** [comment]
Hey! This code was only tested using NVIDIA GPUs, I believe they have a guide for PyTorch here: https://developer.apple.com/metal/pytorch/ :)

**Comment by junhua** [comment]
> Hey! This code was only tested using NVIDIA GPUs, I believe they have a guide for PyTorch here: https://developer.apple.com/metal/pytorch/ :)

Created a [pull request](https://github.com/SakanaAI/AI-Scientist/pull/57) to address this

**Comment by cvanvlack** [comment]
I checked out the code from the pull request and tried to run it. It's been going most of the weekend on my Mac. :)

Specifically on these commands

```bash
# Prepare NanoGPT data
python data/enwik8/prepare.py
python data/shakespeare_char/prepare.py
python data/text8/prepare.py
```

Do these even need to be run? I am curious what is the actual minimal command set of most of the AI work is actual done through OpenAI, etc...

**Comment by conglu1997** [comment]
These commands run the baseline which should be machine dependent for things like training speed!

**Comment by cvanvlack** [comment]
Right, but I am not using a machine with a GPU. I want to offload as much of this as possible to a third-party tool (Colab? Some other service that has a GPU?).

So the fact that this dependency exists is kind of a blocker for usage.

Maybe there's something else I am missing about what this is for?

**Comment by conglu1997** [comment]
I think modern machine learning is quite hard without a GPU. Later experiments will attempt dozens of runs which could take hours each without a GPU. I would recommend services like lambda where you can rent GPUs per hour.

**Comment by mruckman1** [comment]
Why not use GPT4o-mini/Claude instead of a local nanogpt? Not totally sure of the value here for a hybrid approach given the cost of cutting off mac users since we're already providing API keys. .

**Comment by cvanvlack** [comment]
Additionally, is it REALLY a dependency? It looks like it creates an artifact that is used later when you’re using the GPT4o-mini. 

Is that artifact actually a requirement for subsequent steps?

**Comment by conglu1997** [comment]
> Why not use GPT4o-mini/Claude instead of a local nanogpt? Not totally sure of the value here for a hybrid approach given the cost of cutting off mac users since we're already providing API keys. .

GPT4o/Claude is the foundation model that proposes ideas. NanoGPT is the actual model that is modified and trained. This is a different model for different templates.

**Comment by conglu1997** [comment]
> Additionally, is it REALLY a dependency? It looks like it creates an artifact that is used later when you’re using the GPT4o-mini.
> 
> Is that artifact actually a requirement for subsequent steps?

The preparation steps create the training data and baseline for comparison. Very much essential.
